internal/pkg/service/buffer/worker/distribution/listener_test.go

```diff
@@ -43,19 +41,28 @@ func TestOnChangeListener(t *testing.T) {
 			case events := <-listener.C:
 				for _, event := range events {
 					_, _ = listenerLogs.WriteString(fmt.Sprintf("distribution changed: %s\n", event.Message))
-					count.Inc()
 				}
 			}
 		}
 	}()
 
-	// Add node 2,3, stop node 2
+	// Add node 2
 	_, d2 = createNode(t, clk, nil, etcdNamespace, "node2")
+	assert.Eventually(t, func() bool {
+		return strings.Contains(listenerLogs.String(), `found a new node "node2"`)
+	}, time.Second, 10*time.Millisecond, "timeout")
+
+	// Add node 3
 	_, d3 = createNode(t, clk, nil, etcdNamespace, "node3")
+	assert.Eventually(t, func() bool {
+		return strings.Contains(listenerLogs.String(), `found a new node "node3"`)
+	}, time.Second, 10*time.Millisecond, "timeout")
+
+	// Stop node 2
 	d2.Process().Shutdown(errors.New("test"))
 	d2.Process().WaitForShutdown()
 	assert.Eventually(t, func() bool {
-		return count.Load() == 3
+		return strings.Contains(listenerLogs.String(), `the node "node2" gone`)
```

Comment:
Spravny count neznamena, ze sa to stihlo aj zalogovat, ... tak kontrolujeme radsej priamo log msg.